src/dotnet-scaffolding/dotnet-scaffold/Telemetry/CommandExecuteTelemetryEvent.cs

```diff
@@ -15,7 +15,7 @@ public CommandExecuteTelemetryEvent(DotNetToolInfo dotnetToolInfo, CommandInfo c
         SetProperty("Version", dotnetToolInfo.Version);
         SetProperty("ToolName", dotnetToolInfo.Command);
         SetProperty("ToolLevel", dotnetToolInfo.IsGlobalTool ? TelemetryConstants.GlobalTool : TelemetryConstants.LocalTool);
-        SetProperty("CommandName", commandInfo.Name);
+        SetProperty("CommandName", commandInfo.Name.Hash());
```

Comment:
I wonder if we should do something like UA has where we have an overload to SetProperty that takes bool isPii (or make it have a default false and no overload I guess) and it handles the hashing under the hood. So we don't need all the events to know about how we do the hashing, just in case it needs to change. That of course complicates my other comment about AllCommandCategories, but maybe that's the rare case where we can't (or we support a SetProperty overload that takes an IEnumerable<string>).

Reply:
I like this idea, will take a quick peek at UA and having the overload for IEnumerable sounds good.

Reply:
done, verified